tests/test_transform.py

```diff
@@ -180,10 +296,126 @@ def checkTransform(self, fromName, toName):
+                assert_allclose(rawInversePoint, frameSet.tranInverse(rawOutPoint), err_msg=msg)
+
+                # forward transformation of an array of points
+                nPoints = 7  # arbitrary
+                rawInArray = makeRawArrayData(nPoints, nIn)
+                inArray = fromEndpoint.arrayFromData(rawInArray)
+                outArray = transform.tranForward(inArray)
+                rawOutArray = toEndpoint.dataFromArray(outArray)
+                self.assertFloatsAlmostEqual(rawOutArray, polyMap.tranForward(rawInArray), msg=msg)
+                self.assertFloatsAlmostEqual(rawOutArray, frameSet.tranForward(rawInArray), msg=msg)
+
+                # inverse transformation of an array of points;
+                # remember that the inverse will not give the original values!
+                inverseArray = transform.tranInverse(outArray)
+                rawInverseArray = fromEndpoint.dataFromArray(inverseArray)
+                self.assertFloatsAlmostEqual(rawInverseArray, polyMap.tranInverse(rawOutArray), msg=msg)
+                self.assertFloatsAlmostEqual(rawInverseArray, frameSet.tranInverse(rawOutArray), msg=msg)
+
+                # If frameSet has a SkyFrame as base or current, try permuting that (in place);
+                # the Transform constructor should undo the permutation.
+                # Warning: you must permute the axes of the SkYFrame once it is in the frameSet,
+                # rather than before adding it to the FrameSet, else the behavior is wrong
+                for permFrameSet, isBasePermuted, isCurrPermuted in permFrameSetIter(frameSet):
+                    if isBasePermuted:
+                        self.assertEqual(permFrameSet.getFrame(astshim.FrameSet.BASE).getLonAxis(), 2)
+                    if isCurrPermuted:
+                        self.assertEqual(permFrameSet.getFrame(astshim.FrameSet.CURRENT).getLonAxis(), 2)
+                    transformBasePerm = transformClass(permFrameSet)
+                    # the base frame set in the transform should NOT be permuted
+                    permFrameSetCopy = transformBasePerm.getFrameSet()
+                    if fromName == "SpherePoint":   # it is a SkyFrame and so has getLonAxis
+                        self.assertEqual(permFrameSetCopy.getFrame(astshim.FrameSet.BASE).getLonAxis(), 1)
+                    if toName == "SpherePoint":   # it is a SkyFrame and so has getLonAxis
+                        self.assertEqual(permFrameSetCopy.getFrame(astshim.FrameSet.CURRENT).getLonAxis(), 1)
+
+                    # reconstruct transform, just in case
+                    frameSet = makeFrameSet(baseFrame, currFrame)
+                    transform = transformClass(frameSet)
+                    fromEndpoint = transform.getFromEndpoint()
+
+                    # the transform should work identically to `transform`
+                    nPoints = 7  # arbitrary
```

Comment:
Yes. Done as part of the earlier refactoring. I'm really glad you suggested that. It was a huge help.